src/testing/type_utils.rs

```diff
@@ -0,0 +1,25 @@
+use crate::business_logic::execution::{
+    error::ExecutionError,
+    objects::{CallInfo, Event, L2toL1MessageInfo, TransactionExecutionInfo},
+};
+
+pub(crate) enum ExecutionInfo {
+    Transaction(Box<TransactionExecutionInfo>),
+    Call(Box<CallInfo>),
```

Comment:
Do you need the `Box<_>` wrapper?

Reply:
Yes because both structures differ a lot in size, so clippy suggest it to avoid possible errors. It think it makes sense, since in the deep enums are like C unions, if both structures are highly different in size, it is more suitable to have a pointer.